### PR TITLE
disable amp backend for pytorch_fx, pytorch quantization. since this …

### PR DESCRIPTION
**Pull request by sywangyi**
…module does not support bf16 input

Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

# What does this PR do?

Fixes # ([issue](https://github.com/huggingface/optimum-intel/issues/51))

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sywangyi** [comment]
@PenghuiCheng please have a check

**Comment by sywangyi** [comment]
@yao-matrix

File: optimum/intel/neural_compressor/trainer.py [hunk 1]
```diff
@@ -630,3 +630,19 @@ def compute_distillation_loss(self, student_outputs, teacher_outputs):
...
+        eval_dataset: Optional[Dataset] = None,
+        ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: str = "eval",
+    ) -> Dict[str, float]:
+        if self.model is None:
+            return None
+        if self.model.config and self.model.config.torch_dtype == "int8":
+            if self.model.config.framework in ["pytorch", "pytorch_fx"] and self.use_cpu_amp:
+                logger.warn(
+                    f"{self.model.config.framework} quantization does't support BFloat16 input, disable cpu_amp half precision backend."
```
Comment (PenghuiCheng) [inline optimum/intel/neural_compressor/trainer.py:645]: How ablout:
logger.warn(
                    f"{self.model.config.framework} quantized model does't support BFloat16 input, disable cpu_amp half precision backend."

**Comment by sywangyi** [comment]
@echarlaix @ftian1 please help review

**Review by PenghuiCheng — APPROVED** [review]

File: optimum/intel/neural_compressor/trainer.py [hunk 2]
```diff
@@ -630,3 +630,19 @@ def compute_distillation_loss(self, student_outputs, teacher_outputs):
...
+        eval_dataset: Optional[Dataset] = None,
+        ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: str = "eval",
+    ) -> Dict[str, float]:
+        if self.model is None:
+            return None
+        if self.model.config and self.model.config.torch_dtype == "int8":
+            if self.model.config.framework in ["pytorch", "pytorch_fx"] and self.use_cpu_amp:
+                logger.warn(
+                    f"{self.model.config.framework} quantized model does't support BFloat16 input, disable cpu_amp half precision backend."
```
Comment (echarlaix) [inline optimum/intel/neural_compressor/trainer.py:645]:
```suggestion
                    f"{self.model.config.framework} quantized model doesn't support BFloat16 input, setting `use_cpu_amp` to False."

```

**Review by echarlaix — COMMENTED** [review]
Thanks for this addition !

File: optimum/intel/neural_compressor/trainer.py [hunk 3]
```diff
@@ -630,3 +630,20 @@ def compute_distillation_loss(self, student_outputs, teacher_outputs):
...
+
+    def evaluate(
+        self,
+        eval_dataset: Optional[Dataset] = None,
+        ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: str = "eval",
+    ) -> Dict[str, float]:
+        if self.model is None:
+            return None
+        if self.model.config and self.model.config.torch_dtype == "int8":
```
Comment (echarlaix) [inline optimum/intel/neural_compressor/trainer.py:642]: I would replace it with :
```suggestion
        if getattr(self.model.config, "torch_dtype ", None) == "int8":
```

**Comment by echarlaix** [comment]
To make sure the quality test passes, you can run [`make style`](https://github.com/huggingface/optimum-intel/blob/main/Makefile#L27) on your branch.

**Review by echarlaix — APPROVED** [review]
Thanks for iterating on this !

**Review by PenghuiCheng — APPROVED** [review]

**Pull request merged**